internal/collections/duration_collection_test.go

```diff
@@ -0,0 +1,31 @@
+// Copyright 2023 Juan Pablo Tosso and the OWASP Coraza contributors
+// SPDX-License-Identifier: Apache-2.0
+
+package collections
+
+import (
+	"strconv"
+	"testing"
+	"time"
+)
+
+func TestDuration(t *testing.T) {
+	d := NewDuration()
+	time.Sleep(100 * time.Millisecond)
```

Comment:
I'd avoid this. Instead I would set a function to retrieve returns the time and mock for the test.